torch_sim/models/metatensor.py

```diff
@@ -0,0 +1,312 @@
+        """
+        super().__init__()
+
+        if model is None:
+            model = "pet-mad"
+
+        if model == "pet-mad":
+            path = "https://huggingface.co/lab-cosmo/pet-mad/resolve/main/models/pet-mad-latest.ckpt"
+            self._model = load_model(path).export()
+        elif model.endswith(".ckpt"):
+            path = model
+            self._model = load_model(path).export()
+        elif model.endswith(".pt"):
+            path = model
+            self._model = load_atomistic_model(path, extensions_path)
+        else:
+            raise ValueError('Model must be a path to a .ckpt/.pt file, or "pet-mad".')
+
+        self._device = device or ("cuda" if torch.cuda.is_available() else "cpu")
+        self._dtype = getattr(torch, self._model.capabilities().dtype)
+        self._model.to(self._device)
+        self._compute_forces = compute_forces
+        self._compute_stress = compute_stress
+        self.neighbor_list_fn = neighbor_list_fn
+        self._memory_scales_with = "n_atoms_x_density"  # for the majority of models
+        self._check_consistency = check_consistency
+        self._requested_neighbor_lists = self._model.requested_neighbor_lists()
+        self._evaluation_options = ModelEvaluationOptions(
+            length_unit="angstrom",
+            outputs={
+                "energy": ModelOutput(
+                    quantity="energy",
+                    unit="eV",
+                    per_atom=False,
+                )
+            },
+        )
+
+    def forward(  # noqa: C901, PLR0915
+        self,
+        state: SimState | StateDict,
+    ) -> dict[str, torch.Tensor]:
+        """Compute energies, forces, and stresses for the given atomic systems.
+
+        Processes the provided state information and computes energies, forces, and
+        stresses using the underlying metatensor model. Handles batched calculations for
+        multiple systems as well as constructing the necessary neighbor lists.
+
+        Args:
+            state (SimState | StateDict): State object containing positions, cell,
+                and other system information. Can be either a SimState object or a
+                dictionary with the relevant fields.
+
+        Returns:
+            dict[str, torch.Tensor]: Dictionary containing:
+                - 'energy': System energies with shape [n_systems]
+                - 'forces': Atomic forces with shape [n_atoms, 3] if compute_forces=True
+                - 'stress': System stresses with shape [n_systems, 3, 3] if
+                    compute_stress=True
+        """
+        # Extract required data from input
+        if isinstance(state, dict):
+            state = SimState(**state, masses=torch.ones_like(state["positions"]))
+
+        # Input validation is already done inside the forward method of the
+        # MetatensorAtomisticModel class, so we don't need to do it again here.
+
+        atomic_numbers = state.atomic_numbers
+        cell = state.cell
```

Comment:
I think this likely needs to be updated to `state.row_vector_cell`, we use column vector convention similar to LAMMPS rather than rows like ASE.

Reply:
Thanks! Good catch